ts/features/itwallet/credentials/store/selectors/index.ts

```diff
@@ -64,12 +68,50 @@ export const itwIsWalletEmptySelector = createSelector(
   ({ credentials }) => credentials.length === 0
 );
 
+/**
+ * Get the credential status and the error message corresponding to the status attestation error, if present.
+ * The message is dynamic and extracted from the issuer configuration.
+ *
+ * Note: the credential type is passed as second argument to reuse the same selector and cache per credential type.
+ */
+export const itwCredentialStatusSelector = createSelector(
+  itwCredentialsByTypeSelector,
+  (_: GlobalState, type: string) => type,
+  (credentials, type) => {
```

Comment:
What about this refactoring?

```suggestion
export const itwCredentialStatusSelector = (credentialType: string) => createSelector(
  itwCredentialsByTypeSelector,
  (credentialsByType) => {
```

and then use it with:

```js
  const { status } = useIOSelector(itwCredentialStatusSelector(credentialType));
  ```
  
 I think it's easier to read (you are creating a status selector for a specific credential type) and to use. Wha do you think?

Reply:
I chose that approach to reuse the same selector and its cache. This way if a component uses the selector and requests the same credential type `reselect` will return the cached value.

The other way creates a new selector for every component that uses it, so `reselect` won't reuse the cached result.

Probably it's an over-optimization, but I didn't like the same selector executed multiple times when used in several components.

Reply:
It makes sense. I don't think this is over optimization, the selector is doing a lot of things, so a cached value might be a better option!
Thank you!